### PR TITLE
Better handle multiple golden chunks in TPU train

### PR DESCRIPTION
**Pull request by sethtroisi**

### CODE REVIEW

File: preprocessing.py [hunk 1]
```diff
@@ -205,17 +205,20 @@ def get_input_tensors(batch_size, tf_records, num_repeats=1,
 
 
 def get_tpu_input_tensors(batch_size, tf_records, num_repeats=1,
-                          shuffle_records=True, shuffle_examples=True,
-                          shuffle_buffer_size=1024,
                           filter_amount=1, random_rotation=True):
+    # TPUs trains on sequential golden chunks to simplify preprocessing and
+    # reproducability.
```
Comment (brilee) [inline preprocessing.py]: typo:reproducibility

File: preprocessing.py [hunk 2]
```diff
@@ -119,25 +119,24 @@ def read_tf_records(batch_size, tf_records, num_repeats=1,
...
-                                         tf.data.TFRecordDataset(
-                                             x, compression_type='ZLIB'),
-                                         cycle_length=64, block_length=16)
+        dataset = record_list.flat_map(map_func)
+
     if filter_amount < 1.0:
-        dataset = dataset.filter(lambda x: tf.less(
-            tf.random_uniform([1]), filter_amount)[0])
+        dataset = dataset.filter(
+            lambda _: tf.random_uniform([]) < filter_amount)
```
Comment (brilee) [inline preprocessing.py]: does this actually do shape inference to find the right size for tf.random_uniform? That seems pretty cool. 

Curious if this also fixes #520

Comment (sethtroisi) [reply preprocessing.py]: I think it does something different than you expect.
[] is a scalar random and filter get's run on every item in the dataset.

---
```
x = np.random.sample((10, 2)) 
dataset = tf.data.Dataset.from_tensor_slices(x)
iter = dataset.make_one_shot_iterator().get_next()
with tf.Session() as sess:
  while True:
    print(sess.run(iter))
iter = dataset.filter(lambda x: tf.random_uniform([]) < .3).make_one_shot_iterator().get_next()
with tf.Session() as sess:
  while True:
    print(sess.run(iter))
```

Comment (brilee) [reply preprocessing.py]: oh, okay - never mind then :)

File: preprocessing.py [hunk 3]
```diff
@@ -205,17 +205,20 @@ def get_input_tensors(batch_size, tf_records, num_repeats=1,
 
 
 def get_tpu_input_tensors(batch_size, tf_records, num_repeats=1,
-                          shuffle_records=True, shuffle_examples=True,
-                          shuffle_buffer_size=1024,
                           filter_amount=1, random_rotation=True):
+    # TPUs trains on sequential golden chunks to simplify preprocessing and
```
Comment (amj) [inline preprocessing.py]: This isn't really necessarily true after we do Cloud BT integration.  i think you axe this assert and comment.

Comment (sethtroisi) [reply preprocessing.py]: BT doesn't use this code path, it calls `get_tpu_bt_input_tensors` which makes a supcall to bigtable_input to do most of it's processing.

**Pull request merged**